ledger/src/blockstore.rs

```diff
@@ -1138,19 +1108,24 @@ impl Blockstore {
             e.num_repaired += 1;
         }
 
-        // Should be safe to modify index_meta here. Two cases
-        // 1) Recovery happens: Then all inserted erasure metas are removed
-        // from just_received_coding_shreds, and nothing will be committed by
-        // `check_insert_coding_shred`, so the coding index meta will not be
-        // committed
-        index_meta.coding_mut().set_present(shred_index, true);
+        // insert coding shred into rocks
+        let result = self
+            .insert_coding_shred(index_meta, &shred, write_batch)
+            .is_ok();
+
+        if result {
+            index_meta.coding_mut().set_present(shred_index, true);
```

Comment:
this line is redundant:
https://github.com/solana-labs/solana/pull/21214#discussion_r752583065

Reply:
I think we still need it since this line exists in the original `insert_coding_shred`, and this PR merged `cache_coding_shred` and `insert_coding_shred` functions together (i.e. we remove the one in `cache_coding_shred` but keep the one in `insert_coding_shred` so we keep one at the end?)

https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore.rs#L1301